components/gitpod-db/src/typeorm/migration/1662491259313-WorkspaceInstanceStartedStoppingIndex.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { indexExists } from "./helper/helper";
+
+export class WorkspaceInstanceStartedStoppingIndex1662491259313 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const TABLE_NAME = "d_b_workspace_instance";
+
+        const startedTimeIndex = "IDX_workspace_instance__started_time";
+        if (!(await indexExists(queryRunner, TABLE_NAME, startedTimeIndex))) {
+            await queryRunner.query(`CREATE INDEX ${startedTimeIndex} ON ${TABLE_NAME} (startedTime)`);
```

Comment:
Do we need the magical `, ALGORITHM=INPLACE, LOCK=NONE` statement here?

@geropl

Reply:
> magical

Link to MySQL docs: https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl-operations.html

tl;dr: no, adding 2ndary indexes is a in-place operation that does not lock.

This is more relevant for adding primary keys, or re-naming (cmp. docs).